metricflow/sql_clients/databricks.py

```diff
@@ -102,31 +113,47 @@ def _engine_specific_dry_run_implementation(self, stmt: str, bind_params: SqlBin
         """Check that query will run successfully without actually running the query, error if not."""
         self._engine_specific_execute_implementation(stmt=f"EXPLAIN {stmt}", bind_params=bind_params)
 
+    # TODO: this is VERYYY SLOWWWWW.
     def create_table_from_dataframe(  # noqa: D
         self, sql_table: SqlTable, df: pd.DataFrame, chunk_size: Optional[int] = None
     ) -> None:
         logger.info(f"Creating table '{sql_table.sql}' from a DataFrame with {df.shape[0]} row(s)")
         start_time = time.time()
         with self.get_connection() as connection:
-            pd.io.sql.to_sql(
-                frame=df,
-                name=sql_table.table_name,
-                con=connection,
-                schema=sql_table.schema_name,
-                index=False,
-                if_exists="fail",
-                method="multi",
-                chunksize=chunk_size,
-            )
+            with connection.cursor() as cursor:
+                pyarrow_df = pyarrow.Table.from_pandas(
+                    df
+                )  # TODO: if this piece is slow, trash it and use pandas dtypes
+
+                # Create table
+                columns = [col for col in pyarrow_df.itercolumns()]
+                column_names = pyarrow_df.column_names
+                columns_to_insert = []
+                for i in range(pyarrow_df.num_columns):
+                    # Format as "column_name column_type"
+                    columns_to_insert.append(f"{column_names[i]} {PYARROW_TO_SQL_DTYPES[str(columns[i].type)]}")
+                cursor.execute(f"CREATE TABLE IF NOT EXISTS {sql_table.sql} ({', '.join(columns_to_insert)})")
+
+                # Insert rows
+                values = []
+                for row in df.itertuples(index=False, name=None):
+                    cells = []
+                    for cell in row:
+                        if type(cell) in [str, pd.Timestamp]:
+                            # Wrap cell in quotes & escape existing single quotes
+                            cells.append(f"""'{str(cell).replace("'", '"')}'""")
+                        else:
+                            cells.append(str(cell))
+                    values.append(f"({', '.join(cells)})")
+                cursor.execute(f"INSERT INTO {sql_table.sql} VALUES {', '.join(values)}")
```

Comment:
@courtneyholcomb - We've had some issues in the past where without `chuck_size` support, we would get an error when loading larger data frames. Nit: mind breaking the values into separate lines? This could make it hard to read in the query logs and some chance it causes memory issues in Databricks if their parsing implementation is off.